api/v1/health/server/server.go

```diff
@@ -318,21 +384,31 @@ func (s *Server) Serve() (err error) {
 		configureServer(httpsServer, "https", s.httpsServerL.Addr().String())
 
 		servers = append(servers, httpsServer)
-		wg.Add(1)
+		s.wg.Add(1)
 		s.Logf("Serving cilium health API at https://%s", s.httpsServerL.Addr())
 		go func(l net.Listener) {
-			defer wg.Done()
+			defer s.wg.Done()
 			if err := httpsServer.Serve(l); err != nil && err != http.ErrServerClosed {
 				s.Fatalf("%v", err)
 			}
 			s.Logf("Stopped serving cilium health API at https://%s", l.Addr())
 		}(tls.NewListener(s.httpsServerL, httpsServer.TLSConfig))
 	}
 
-	wg.Add(1)
-	go s.handleShutdown(wg, &servers)
+	s.wg.Add(1)
+	go s.handleShutdown(&servers)
+
+	return nil
+}
 
-	wg.Wait()
+func (s *Server) Stop(hive.HookContext) error {
+	if err := s.Shutdown(); err != nil {
+		return err
+	}
+	// FIXME need to pass HookContext somehow into handleShutdown to make this
```

Comment:
What would be the implication of not fixing this FIXME? A call to `.Stop()` would not respect an intended grace period?

Reply:
We call https://pkg.go.dev/net/http#Server.Shutdown on shutdown which waits indefinitely for existing connections to go away. If there's a blocking connection we may end up exceeding the default 5 minute pkg/hive stop timeout and fatal and thus end up with an unclean shutdown. Currently this isn't really much of an issue, but it is worth fixing.

To fix this, I'd need to refactor the code a bit to use the stop context to call `server.Shutdown(ctx)` and if it fails with cancelled context I'd forcefully close the server with `Close()`. Since this isn't a huge amount of work I'll just do this now.